inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -2859,14 +2990,105 @@ void MKLDNNInterpolateNode::linearOnnxRef(const uint8_t *in_ptr_, uint8_t *out_p
+                int xInRange = std::max(0, std::min(x, IW - 1));
+                retX += xFactor[ox * CUBIC_GRID_LEN + j] * in_ptr_nch[xInRange];
+            }
+            retY += yFactor[oy * CUBIC_GRID_LEN + i] * retX;
+        }
+        out_ptr_nc[oy * OW + ox] = retY;
+    });
+}
+
+float MKLDNNInterpolateNode::InterpolateRefExecutor::getValue(const uint8_t *base, size_t offset, InferenceEngine::Precision prec) {
+    const uint8_t *baseOffset = base + offset;
+    switch (prec) {
+        case Precision::U8: {
+            return static_cast<float>(*baseOffset);
+            break;
+        }
+        case Precision::I8: {
+            const int8_t *valuePtr = reinterpret_cast<const int8_t *>(baseOffset);
+            return static_cast<float>(*valuePtr);
+            break;
+        }
+        case Precision::BF16: {
+            const uint16_t *valuePtr = reinterpret_cast<const uint16_t *>(baseOffset);
+            return bfloat16_t::from_bits(*valuePtr);
+            break;
+        }
+        case Precision::FP32: {
+            const float *valuePtr = reinterpret_cast<const float *>(baseOffset);
+            return *valuePtr;
+            break;
+        }
+        default: {
+            IE_THROW() << "Interpolate layer does not support precision: " << prec;
+            break;
+        }
+    }
+}
+
+void MKLDNNInterpolateNode::InterpolateRefExecutor::setValue(uint8_t *base, size_t offset, float value, InferenceEngine::Precision prec) {
+    uint8_t *baseOffset = base + offset;
+    switch (prec) {
+        case Precision::U8: {
+            uint8_t data = static_cast<uint8_t>(value < 0 ? 0 : value);
+            cpu_memcpy(baseOffset, &data, 1);
+            break;
+        }
+        case Precision::I8: {
+            int8_t data = static_cast<int8_t>(value);
+            cpu_memcpy(baseOffset, &data, 1);
+            break;
+        }
+        case Precision::BF16: {
+            uint16_t data = bfloat16_t(value).to_bits();
+            cpu_memcpy(baseOffset, &data, 2);
+            break;
+        }
+        case Precision::FP32: {
+            cpu_memcpy(baseOffset, &value, sizeof(float));
+            break;
+        }
+        default: {
+            IE_THROW() << "Interpolate layer does not support precision: " << prec;
+            break;
+        }
+    }
 }
 
-void MKLDNNInterpolateNode::linearInterpolation(const uint8_t *in_ptr_, uint8_t *out_ptr_, int B, int C, int ID, int IH, int IW,
+void MKLDNNInterpolateNode::InterpolateRefExecutor::linearInterpolation(const uint8_t *in_ptr_, uint8_t *out_ptr_, int B, int C, int ID, int IH, int IW,
                                           float fx, float fy, float fz, int OD, int OH, int OW, int kernel_width, bool antialias) {
     if (IW == OW && IH == OH && ID == OD) {
         size_t spatialDimSize = IW * IH * ID;
-        if (fusedWith.empty() && inputPrec == outputPrec) {
+        // TODO: enable when fusing into interp with linear mode will support
```

Comment:
At this moment we doesn't support fusing in Interpolate with linear mode or if `ISA < cpu::x64::sse41`
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp#L3226
So this check is useless since we execute Interpolate only as Reference.
I guess we no need to store `fusedWith` inside `InterpolateRefExecutor`